modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+
+To stream successfully, it is important to understand the following:
+
+* The basic unit of the stream is specific to the data format. The unit is a record in a CSV document, an element of an array in a JSON document, or a collection in an XML document.
+* Streaming accesses each unit of the stream sequentially. Streaming does not support random access to a document.
+
+== Enable Streaming
+
+Streaming is not enabled by default. You can use two configuration properties to stream data in a supported data format:
+
+* `streaming` property for reading source data as a stream
+* `deferred` writer property for passing an output stream directly to the next message processor in a flow
+
+To enable streaming when reading source data, you must set the `streaming` reader property to `true` on the data source. You can enable streaming on a data source through the *MIME Type* setting `outputMimeType`. You can set the property in a Mule component that accepts source data, such as an HTTP listener or an On New or Updated File operation.
+
+[source,XML,linenums]
+----
+<flow name="dw-streaming-example" >
+  <http:listener doc:name="Listener"
+      outputMimeType="application/json; streaming=true"
+      config-ref="HTTP_Listener_config" path="/input"/>
+</flow>
+----
+
+Notice that `streaming=true` is part of the `outputMimeType` value. Other Mule components, such as the File and FTP components, also support the MIME Type setting.
+
+To pass the streamed output to the next message processor, you can use the `output` directive in a DataWeave script, for example:
+
+[source,XML,linenums]
+----
+output application/json deferred=true
+----
+
+For more detail on use of the `deferred` property, see <<stream_output>>.
+
+[[stream_csv]]
+== Streaming CSV
+
+CSV is simplest format for streaming because of its structure. Each row below the CSV header is a streamable record. The following CSV example consists of records that contain `name`, `lastName`, and `age` values.
+​
+[source,CSV,linenums]
+----
+name,lastName,age
+mariano,achaval,37
+leandro,shokida,30
+pedro,achaval,4
+christian,chibana,25
+sara,achaval,2
+matias,achaval,8
+----
+​
+To stream the CSV example, the following script selects values from each record. It uses the `map` function to iterate over each record in the document.
+​
+[source,DataWeave,linenums]
+----
+payload map (record) ->
+    {
+        FullName: record.lastName ++ "," ++ record.name,
+        Age: record.age
+    }
+----
+
+Although streaming does not support random access to the entire document, a DataWeave script can access data randomly _within each record_ because each record is loaded into memory. For example, the expression `record.lastName ++ "," ++ record.name,` can access a `lastName` value before it accesses a `name` value even though the order of values is reversed in the input.
```

Comment:
No we don't need to explain it. The answer is also implied in the next sentence. Their script can access the fields in the record in any order because the record is stored in memory, whereas the entire doc is not.